util/dynamic_drf_api.py

```diff
@@ -0,0 +1,132 @@
+from allocations.allocations_api import BalanceServiceClient
+from allocations.models import Allocation
+from django.contrib.auth import get_user_model
+from dynamic_rest.serializers import DynamicModelSerializer
+from dynamic_rest.viewsets import DynamicModelViewSet
+from projects.models import Field, Project, Type
+
+
+class UserSerializer(DynamicModelSerializer):
+    """Serialize user object."""
+
+    class Meta:
+        """default properties to serialize."""
+
+        model = get_user_model()
+
+        fields = [
+            "username",
+            "first_name",
+            "last_name",
+            "email",
+        ]
+
+
+class UserViewSet(DynamicModelViewSet):
+    """Project field names API."""
+
+    queryset = get_user_model().objects.all()
+    serializer_class = UserSerializer
+
+
+class AllocationSerializer(DynamicModelSerializer):
+    """Serialize Allocation object."""
+
+    """
+    keys in the array "fields" can be queried in URL by "?key=value"
+    Their format can be overridden by defining it outside the "Meta" class.
+    """
+
+    # if self.instance.status == "active":
```

Comment:
Yeah, I was going to ask how this was going to work. Is it possible to make a single batch fetch to the balance service as part of serializing the allocations?

Reply:
The need to do a bulk fetch seems dependent on triggering the "list view".

One way to do it would be to add to the view, and have it pass the su_usage to the serializer. 
https://www.django-rest-framework.org/api-guide/serializers/#including-extra-context

Alternately, we could have a celery task to just periodically update the db?